sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -177,7 +177,7 @@ public void forceFlush() {
     private final AtomicBoolean exportAvailable = new AtomicBoolean(true);
 
     @GuardedBy("monitor")
-    private final List<ReadableSpan> spansList;
+    private final ArrayList<ReadableSpan> spansList;
```

Comment:
Exposing an array list here as the operations we perform on it assumes an array in terms of bulk copying and therefore, o(1) element access.